pyjelly/integrations/rdflib/parse.py

```diff
@@ -466,8 +395,11 @@ def parse(self, source: InputSource, sink: Graph) -> None:
         if inp is None:
             msg = "expected source to be a stream of bytes"
             raise TypeError(msg)
-        graph_from_jelly(
+        parse_jelly_flat(
             inp,
-            identifier=sink.identifier,
-            store=sink.store,
+            sink=sink,
+            lambda_graph=lambda store, identifier: Graph(
+                store=store, identifier=identifier
+            ),
+            lambda_dataset=lambda store: Dataset(store=store),
```

Comment:
instead write

`lambda: Dataset(store=sink.store)`

And all your problems are solved :)